src/createRestyleFunction.ts

```diff
@@ -24,14 +45,58 @@ const createRestyleFunction = <
     props,
     {theme, dimensions},
   ) => {
-    const value = getResponsiveValue(props[property], {
-      theme,
+    // We reuse the theme object for saving the memoized values.
+    const unsafeTheme: ThemeWithMemoization<Theme> =
+      theme as ThemeWithMemoization<Theme>;
+    if (unsafeTheme.unsafeMemoizedMap == null) {
+      unsafeTheme.unsafeMemoizedMap = {};
+    }
+
+    let memoizedMapHashKey;
+
+    memoizedMapHashKey = getMemoizedMapHashKey(
       dimensions,
-      themeKey,
-      transform,
-    });
+      String(themeKey),
+      String(property),
+      String(props[property]),
+    );
```

Comment:
we are calling the method twice, I would only call it if we see that `themeKey` and `property` are not strings.